common/util/vector_tree.h

```diff
@@ -316,6 +314,9 @@ class VectorTree : private _VectorTreeImpl {
     return *this;
   }
 
+  // Hint the implementation how many children are to be expected.
+  void SetExpectedChildrenUpperBound(size_t s) { children_.reserve(s); }
```

Comment:
could just name it something like `ReserveChildren()`?

Reply:
I wanted to make the intention clear: that we provide an upper bound.
`Reserve` sounds too close to the actual implementation of the idea.